packages/components/src/components/SquigglePlayground/LeftPlaygroundPanel/DependencyGraphModal.tsx

```diff
@@ -0,0 +1,25 @@
+import { FC, lazy } from "react";
+
+import { SqProject } from "@quri/squiggle-lang";
+
+const Mermaid = lazy(() => import("../../ui/Mermaid.js"));
+
+export const DependencyGraphModal: FC<{
+  project: SqProject;
+}> = ({ project }) => {
+  const sourceIds = project.getSourceIds();
+
+  let diagram = "graph TD\n";
+  for (const sourceId of sourceIds) {
+    diagram += `${sourceId}\n`; // for sources that don't have any dependencies
+    for (const dependencyId of project.getDependencies(sourceId)) {
+      diagram += `${sourceId} --> ${dependencyId}\n`;
+    }
+  }
```

Comment:
This seems neat, though I would have preferred it being a different story. (Though maybe you wanted this to help with development)

Reply:
Yeah, mostly I wanted to check how sources and dependencies are updated on project changes (there are tests in squiggle-lang, but they're less transparent).

Reply:
yep, makes sense